src/style-guide.scrbl

```diff
@@ -188,6 +190,35 @@ fun sqrt(n :: Number%(non-negative)) -> Number:
 end
 }
 
+@subsection{Parametric Annotations}
+
+You should avoid using @tt{Any} as the subtype of a parametric data structure,
+like a @tt{List}.
+
+In general, a @tt{List<Any>} definition is often undesirable because it
+indicates elements of any type can be members of the list. In particular,
+this subtly implies that multiple elements of different types could exist.
+In other words, @tt{[list: 1, "apple", true]} would satisfy the annotation
+@tt{List<Any>}. Creating @tt{List}s like this is usually bad practice.
+
+In Pyret, we can do this by parametrizing the type. Rather than
+
+@codedisp{
+fun search(lst :: List<Any>) -> Any: ... end
+}
+
+we'd now do something like
+
+@codedisp{
+fun search<A>(lst :: List<A>) -> A: ... end
+}
+
+Now, the elements of @tt{lst} can be of any type, but all elements of lst will
```

Comment:
Bad wording -- you just got through saying you don't want to have a list of Any type, but it's ok now to have a list of any type?  (Note the capitalization...)  You mean to say something closer to "Now, we don't know or care what the type of the elements of `lst` are, but we are asserting that they all be of the *same* type -- namely `A`.  This distinction (between "any type at all", and "any single type, for any particular list") is subtle but important."